runtime/doc/lua.txt

```diff
@@ -1888,28 +1983,28 @@ ringbuf({size})                                                *vim.ringbuf()*
     Return: ~
         (table)
 
-Ringbuf:clear({self})                                        *Ringbuf:clear()*
+vim.Ringbuf:clear()                                          *Ringbuf:clear()*
```

Comment:
This isn't ideal.

I tried special casing this so method exclude the `vim.` prefix, but that causes the functions to be moved far away from `vim.ringbuf`. Either:

- `vim.ringbuf` goes in its own module
- We remove `vim.` from all functions

Reply:
> Either:
> 
> * `vim.ringbuf` goes in its own module
> * We remove `vim.` from all functions

The first option sounds much lower cost, for this one case. But does that mean all "classes" need to have dedicated modules? I hope not...

Reply:
Depends how smart we can make `gen_vimdoc.py` 😉 

It's getting pretty complicated already however.